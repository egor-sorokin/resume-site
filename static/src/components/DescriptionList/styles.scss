.description-list {
  margin-top: 10px;
  width: 100%;

  .item {
    overflow: hidden;
  }

  .item:nth-of-type(n+1) {
    margin-top: 20px;
  }

  .item__plain-text {
    line-height: 1.3;
  }
}

