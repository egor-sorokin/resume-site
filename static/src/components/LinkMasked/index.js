import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const LinkStretched = ({linkData, cssClasses, target}) => (
  <div className="link-masked">
    <span className={`link-masked__placeholder ${cssClasses}`}>{linkData.text}</span>
    <div className="link-masked__mask">
      <a
        href={linkData.url}
        target={target}
        className={`link-masked__text ${cssClasses}`}
      >
        {linkData.text}
      </a>
    </div>
  </div>
);

LinkStretched.propTypes = {
  linkData: PropTypes.shape({}),
  cssClasses: PropTypes.string,
  target: PropTypes.string
};

LinkStretched.defaultProps = {
  linkData: {},
  cssClasses: '',
  target: ''
};

export default LinkStretched;