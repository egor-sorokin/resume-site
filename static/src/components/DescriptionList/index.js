import React from 'react';
import PropTypes from 'prop-types';
import LinkMasked from '../LinkMasked/index';
import './styles.scss';


const DescriptionList = ({listData}) => {
  const cssClassesLinkMasked = "link-masked--white font-s-20 text-c-l-dune";
  const component = listData.map((item, i) => (
    <li
      key={i+1}
      className="description-list__item item">
      <h6 className="item__title font-s-12-secondary">
        {item.label}
      </h6>
      {item.url ? (
        <div className="item__link">
          <LinkMasked
            linkData={item}
            cssClasses={cssClassesLinkMasked}
            target="_blank"
          ></LinkMasked>
        </div>
      ) : (
        <p className="item__plain-text font-s-20 text-c-l-dune">
          {item.text}
        </p>
      )}
    </li>
  ));

  return (
    <ul className="description-list">
      {component}
    </ul>
  );
};

DescriptionList.propTypes = {
  listData: PropTypes.instanceOf(Array),
};

DescriptionList.defaultProps = {
  listData: [],
};

export default DescriptionList;
