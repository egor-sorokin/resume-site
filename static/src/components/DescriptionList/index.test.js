// eslint-disable-next-line
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import DescriptionList from './index';
import LinkMasked from '../LinkMasked/index';


describe('<DescriptionList />', () => {
  let component;
  const listData = [
    {
      label: 'Label test',
      text: 'Text test',
      url: ''
    },
    {
      label: 'Label test',
      text: 'Text test',
      url: 'example.url'
    }
  ];


  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<DescriptionList />);
    });


    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });


    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 0 components', () => {
      expect(component.children()).toHaveLength(0);
    });
  });


  describe('test setting props', () => {
    beforeEach(() => {
      component = mount(<DescriptionList />);
    });


    it('set props correctly', () => {
      expect('listData' in component.props()).toEqual(true);
    });
  });


  describe('test props', () => {
    beforeEach(() => {
      component = shallow(
        <DescriptionList listData={listData}/>
      );
    });


    it('renders a plain text', () => {
      expect(component.find('li').children().someWhere(n => n.hasClass('item__plain-text'))).toBeTruthy();
    });


    it('renders LinkMasked', () => {
      expect(component.find(LinkMasked)).toHaveLength(1);
      expect(component.find('li').children().someWhere(n => n.hasClass('item__link'))).toBeTruthy();
    });
  });
});
