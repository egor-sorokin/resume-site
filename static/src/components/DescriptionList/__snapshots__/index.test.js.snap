// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DescriptionList /> base tests matches snapshot 1`] = `
<ul
  className="description-list"
/>
`;
